internal/applications/cache/caching_service_impl.go

```diff
@@ -0,0 +1,90 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-redis/redis/v8"
+	"github.com/pierrec/lz4/v4"
+	"github.com/vmihailenco/msgpack/v4"
+	"time"
+)
+
+type CachingServiceImpl struct {
+	redisClient *redis.Client
+}
+
+func NewCachingServiceImpl(redisClient *redis.Client) *CachingServiceImpl {
+	return &CachingServiceImpl{redisClient: redisClient}
+}
+
+func (c *CachingServiceImpl) Create(ctx context.Context, key string, data interface{}, expiration time.Duration) (bool, error) {
+
+	serializedData, err := msgpack.Marshal(&data)
+	if err != nil {
+		fmt.Println("Failed for marshaling data:", err)
```

Comment:
```suggestion
		log.error("Failed for marshaling data:", err)
```
pakai log nya echo ya

Reply:
done